implementation/src/main/java/io/smallrye/health/SmallRyeHealthReporter.java

```diff
@@ -158,57 +300,58 @@ private final SmallRyeHealth getHealth(Iterable<HealthCheck>... checks) {
         return new SmallRyeHealth(builder.build());
     }
 
-    private HealthCheckResponse.State processChecks(Iterable<HealthCheck> checks, JsonArrayBuilder results,
-            HealthCheckResponse.State status) {
-        if (checks != null) {
-            for (HealthCheck check : checks) {
-                status = fillCheck(check, results, status);
-            }
-        }
-
-        return status;
-    }
-
-    private HealthCheckResponse.State fillCheck(HealthCheck check, JsonArrayBuilder results,
+    private HealthCheckResponse.State handleResponse(HealthCheckResponse response, JsonArrayBuilder results,
             HealthCheckResponse.State globalOutcome) {
-        if (check == null) {
-            return globalOutcome;
-        }
-        JsonObject each = jsonObject(check);
-        results.add(each);
+        JsonObject responseJson = jsonObject(response);
+        results.add(responseJson);
+
         if (globalOutcome == HealthCheckResponse.State.UP) {
-            String status = each.getString("status");
+            String status = responseJson.getString("status");
             if (status.equals("DOWN")) {
                 return HealthCheckResponse.State.DOWN;
             }
         }
+
         return globalOutcome;
     }
 
-    private JsonObject jsonObject(HealthCheck check) {
+    private Uni<HealthCheckResponse> callAsync(AsyncHealthCheck asyncHealthCheck) {
+        return withRecovery(asyncHealthCheck.getClass().getName(), asyncHealthCheck.call());
```

Comment:
You probably want to add a timeout too.

Reply:
You may want to also handle `null` as item (it should be marked as fail I believe).